staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_progress.go

```diff
@@ -103,6 +106,24 @@ func (pr *conditionalProgressRequester) Run(stopCh <-chan struct{}) {
 			if !shouldRequest {
 				continue
 			}
+			// There can exist an extremely unlikely scenario where the progress requester can
+			// repeatedly request progress notifications even when the feature might not be
+			// supported.
+			//
+			// The feature checker returns `true` if any node in an etcd cluster has the feature
+			// enabled. However, it may change this to `false` if it eventually detects a node
+			// which does not have this enabled. Ideally, till we detect `false` we work under the
+			// assumption that this feature is supported, however we should stop requesting progress
+			// notifications as soon as we get a `false` in order to prevent loading etcd with
+			// redundant requests.
+			if !etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress) {
+				pr.mux.Lock()
+				pr.stopped = true
```

Comment:
I'm not sure if we should stop the requestor here, maybe we could just change the shouldRequest function. 

The question is whether we want to check the feature gate in the requestor or simply reserve/delegate the checking of that flag to the higher layers.

/cc @wojtek-t @serathius

Reply:
Yup - this is not a place where we should check it.

We're aware of the issue where it may change, but it was conscious decision and we generally don't expect setups where there are multiple etcd clusters backing a single cluster being in different versions.
So we consciously went for simplicity and accept a corner case we're not handling that should never happen.

Reply:
That makes sense to me @p0lyn0mial @wojtek-t, thanks.

For my curiosity, if we were to revisit this in the future sometime, what would be the right place to check this if not the progress checker? I would have assumed that since it can change at runtime, its better to check it at the place where its periodically requesting the additional request.

/close